src/compile_target.zig

```diff
@@ -0,0 +1,439 @@
+        // Nobody wants to remember whether its "bun-linux-x64" or "bun-x64-linux".
+        var splitter = bun.strings.split(input, "-");
+        while ((!found_arch or !found_os or !found_baseline or !found_version or !found_libc) and input.len > 0) {
+            const token = splitter.next() orelse break;
+            if (token.len == 0) continue;
+
+            if (!found_arch) {
+                if (Environment.Archictecture.names.get(token)) |arch| {
+                    this.arch = arch;
+                    found_arch = true;
+                    continue;
+                }
+            }
+
+            if (!found_os) {
+                if (Environment.OperatingSystem.names.get(token)) |os| {
+                    this.os = os;
+                    found_os = true;
+                    continue;
+                }
+            }
+
+            if (!found_baseline) {
+                // i don't know what else to call this
+                if (strings.eqlComptime(token, "modern")) {
+                    this.baseline = false;
+                    found_baseline = true;
+                    continue;
+                } else if (strings.eqlComptime(token, "baseline")) {
+                    this.baseline = true;
+                    found_baseline = true;
+                    continue;
+                }
+            }
+
+            if (!found_version) {
+                if (strings.hasPrefixComptime(token, "v1.") or strings.hasPrefixComptime(token, "v0.")) {
+                    const version = bun.Semver.Version.parse(bun.Semver.SlicedString.init(token[1..], token[1..]));
+                    if (version.valid) {
+                        this.version = .{
+                            .major = version.version.major.?,
+                            .minor = version.version.minor.?,
+                            .patch = version.version.patch.?,
+                        };
+                        found_version = true;
+                        continue;
+                    }
+                }
+            }
+
+            if (!found_libc) {
+                if (strings.eqlComptime(token, "musl")) {
+                    this.libc = .musl;
+                    found_libc = true;
+                    continue;
+                }
+            }
+        }
+
+        // default to x64 if no arch is specified but OS is specified
+        // On macOS arm64, it's kind of surprising to choose Linux arm64 or Windows arm64
+        if (!found_arch and found_os) {
+            this.arch = .x64;
+            found_arch = true;
+        }
+
+        // there is no baseline arm64.
+        if (this.baseline and this.arch == .arm64) {
+            this.baseline = false;
+        }
+
+        if (this.libc == .musl and this.os != .linux) {
+            Output.errGeneric("invalid target, musl libc only exists on linux", .{});
+            Global.exit(1);
+        }
+
+        if (this.arch == .wasm or this.os == .wasm) {
```

Comment:
im not sure why `wasm` was added to the list of parsable oses if it is just going to create this error message. it's alright i guess but it just seems extra.

Reply:
people are going to try it